src/battle_script_commands.c

```diff
@@ -13217,11 +13218,20 @@ static void Cmd_healpartystatus(void)
             {
                 u16 ability;
 
-                if (gBattlerPartyIndexes[gBattlerAttacker] == i)
+                if (B_HEAL_BELL_SOUNDPROOF == GEN_5
+                 || (i == gBattlerPartyIndexes[gBattlerAttacker] && B_HEAL_BELL_SOUNDPROOF >= GEN_9))
+                    ability = ABILITY_NONE;
+                else if (B_HEAL_BELL_SOUNDPROOF > GEN_5
+                      && i != gBattlerPartyIndexes[gBattlerAttacker]
+                      && !(gBattleTypeFlags & BATTLE_TYPE_DOUBLE
+                      && gBattlerPartyIndexes[battler] == i
+                      && !(gAbsentBattlerFlags & gBitTable[battler])))
```

Comment:
```suggestion
                      && !(gAbsentBattlerFlags & gBitTable[battler])))
                      && IsBattlerAlive(battler)))
```

Reply:
I'm not sure I understand the thought process here.  This would make it use `GetAbilityBySpecies` instead of `GetBattlerAbility` if the battler is fainted.

Reply:
why? I don't get it. You are checking if the mon is on the field. IsBattlerAlive does the same with a few more checks.

Reply:
So `IsBattlerAlive` always fails in doubles if used on the partner positions?

Reply:
why would it fail?

Reply:
Uh i meant in singles